### ISSUE TITLE
gencli: --page_size defaults to zero and short-circuits calls to List methods.

### ISSUE THREAD

**Issue by timburks** [googleapis/gapic-generator-go#315]
While writing a Go gRPC server, I noticed that my List RPCs weren't getting called. My other RPCs were fine, and after lots of staring I remembered that --page_size defaults to zero. That seems to short-circuit calls to the handler in my Go server so that it doesn't even get called. I guess that's a nice optimization, but it's mystifying.

Raising this for discussion, but considering https://aip.dev/158, we may want to request a change to the Go gRPC server. WDYT?

**Comment by noahdietz** [comment]
> we may want to request a change to the Go gRPC server

Yeah that would be idiomatic thing to do.

I wouldn't be opposed to adding some low default, like 5, for page_size fields in gencli. I'd just feel weird about "overriding" the server's documented default behavior.

**Comment by odsod** [comment]
This one keeps causing confusion for first-time users. Our interpretation of [AIP-132](https://aip.dev/132) is that the service chooses a default `page_size` when not specified, e.g. `50` in the common case:

```proto
  // The maximum number of books to return. The service may return fewer than
  // this value.
  // If unspecified, at most 50 books will be returned.
  // The maximum value is 1000; values above 1000 will be coerced to 1000.
  int32 page_size = 2;
```

I think the right thing to do in the CLI generator is to not short-circuit at all, or at least emit a log message that lets the user know that the call was short circuited.

What do you think?

**Comment by noahdietz** [comment]
Yes, your interpretation of AIP-132 is correct.

It seems like we will be limited by Go here. The request message will default to `0` if the field is unset. yes, the flag sets a default (which is `0` right now, [for example](https://github.com/googleapis/gapic-showcase/blob/master/cmd/gapic-showcase/paged-expand.go#L28)), but even if this bit was left out, struct creation would still assign Go's default empty value for `int32` to the field, which is `0`. Therefore, the server sees the request message as `page_size = 0` (unless I'm missing something).

We should set a sane default (like `10`) in the flag constructor and document that in the flag usage.

WRT logging, I think it's reasonable to do, but this exists via `--verbose`. The request message will be printed, including what the `page_size`  field is set to.

If there are no objections, I will implement this "sane default" for page_size fields.

**Comment by noahdietz** [comment]
Released in v0.12.1. Give it a shot please!
